src/rpm/package.rs

```diff
@@ -61,6 +63,94 @@ impl Package {
         self.write(&mut io::BufWriter::new(fs::File::create(path)?))
     }
 
+    /// Iterate over the file contents of the package payload
+    ///
+    /// # Examples
+    /// ```ignore
+    /// # fn main() -> Result<(), Box<dyn std::error::Error>> {
+    /// let package = rpm::Package::open("test_assets/freesrp-udev-0.3.0-1.25.x86_64.rpm")?;
+    /// for entry in package.files()? {
+    ///     let file = entry?;
+    ///     // do something with file.content
+    ///     println!("{} is {} bytes", file.metadata.path.display(), file.content.len());
+    /// }
+    /// # Ok(()) }
+    /// ```
+    pub fn files(&self) -> Result<FileIterator, Error> {
+        let file_entries = self.metadata.get_file_entries()?;
+        let archive = decompress_stream(
+            self.metadata.get_payload_compressor()?,
+            io::Cursor::new(self.content.clone()),
+        )?;
+
+        Ok(FileIterator {
+            file_entries,
+            archive,
+            count: 0,
+        })
+    }
+
+    /// Extract all contents of the package payload to a given directory
+    ///
```

Comment:
I think more detail is required on the behaviour, will it override existing files, absolute vs relative paths and security considerations.